### PR TITLE
Bump Vert.x dependency to 3.9.1

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Maintenance

### Description

This PR updates the Vert.x version to 3.9.1. The main change is that `setHandler(...)` method of `Future` is depdecated and replaced with `onComplete(...)`. These funstions seem to be replacebale one to one without any further changes to the code.

Apart from updating Vert.x, I also renamed the Maven property `vertx-juni5.version` to `vertx-junit5.version`. I assume that was just a typo in the name.

### Checklist

- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -78,8 +78,8 @@
         <fabric8.zjsonpatch.version>0.3.0</fabric8.zjsonpatch.version>
         <okhttp.version>3.12.6</okhttp.version>
         <okio.version>1.15.0</okio.version>
-        <vertx.version>3.8.5</vertx.version>
-        <vertx-juni5.version>3.8.5</vertx-juni5.version>
```
Comment (samuel-hawker) [inline pom.xml:82]: Out of interest, was this just a typo originally or was it commonly know as `juni5` ?

Comment (ppatierno) [reply pom.xml:82]: I would say just a typo.

Comment (scholzj) [reply pom.xml:82]: It is used for `vertx-junit5` - so I assume as just a typo.

**Pull request merged**